### DISCUSSION TITLE
Can't exactly reproduce v1.3 Factors from CalculateEmissionFactors.Rmd

### DISCUSSION THREAD

**Discussion by MoLi7** [USEPA/supply-chain-factors#80] [Q&A] [answered]
Hi Wes and team,

Thank you for releasing the v1.3 Factors. I played with [CalculateEmissionFactors.Rmd](https://github.com/USEPA/supply-chain-factors/blob/v1.3.0/CalculateEmissionFactors.Rmd) on the main branch in order to generate SEFs in producer price, but I noticed that I can't exactly reproduce the published v1.3 Factors.

Here is a snippet of my comparison that shows, e.g. for USEEIO sector 483000, the published SEF (CO2e) is 0.816, but I got 0.772 from executing CalculateEmissionFactors.Rmd.
![v1 3_factors_comparison](https://github.com/USEPA/supply-chain-factors/assets/28787399/1d8ced78-f006-4e53-8d8c-e46ffdf2561b)

Could you help me understand if there is anything I missed here?

Here is the CSV I generated for your reference. [SupplyChainGHGEmissionFactors_v1.3.0_NAICS_CO2e_USD2022_ML.csv](https://github.com/user-attachments/files/16170029/SupplyChainGHGEmissionFactors_v1.3.0_NAICS_CO2e_USD2022_ML.csv)

Appreciate your help 🙏 
Mo

**Comment by WesIngwersen** [comment]
@bl-young please check

**Reply by bl-young** [reply]
Yes I see this issue - I see the same result for 483000 as @MoLi7. I will follow up with you when feasible to discern what may be going on.

**Reply by MoLi7** [reply]
Thank you Ben!

**Reply by bl-young** [reply, marked as answer]
@MoLi7 thanks for checking this - there was an "error" (so to speak) and the factors were generated using a branch of useeior where https://github.com/USEPA/useeior/issues/281 was resolved. As a result, when you run the factors you get slightly lower results because in v1.5.1 of useeior for which this release was aligned, that error is not yet fixed. However, you may note that if you compare the results by flow (the `_byGHG_` file) they should be the same!

TBD the specific plan for how we will resolve this in the code, but stay tuned.

**Reply by MoLi7** [reply]
@bl-young that makes sense, thank you! I can confirm the results by flow in `_byGHG_` file are indeed the same.

**Reply by bl-young** [reply]
@MoLi7 useeior and the supply chain factors have been updated. Please retry when you get a moment.

**Reply by MoLi7** [reply]
Thank you @bl-young, I can confirm that fix does resolve the discrepancies I saw in 483000 and other sectors, but it seemingly introduced a new problem for USEEIO sector 333914 in the CO2e table: the published SEF and MEF are 0.223 and 0, but now they are 0.175 and 0.023. However, I was seeing 0.222 and 0 before the fix (in the spreadsheet attached above). Did the fix accidentally changed what was correct for 333914?

Btw, I can reproduce the exact Factors for all other sectors with the latest code.

**Reply by bl-young** [reply]
Hi Mo, you are correct about that discrepancy for 333914. I have not been able to fully confirm what is causing the change in the SEF. In v1.5.2, some of the BEA objects were revised slightly (when BEA released earlier data vintage), including the margins table (margins were added to 333914 which did not exist prior). The values you are generating (0.175 and 0.023) I believe are correct for that sector.

**Reply by MoLi7** [reply]
That makes sense! I remember 333914 - it used to be 33391A in 2012 BEA schema, it's not surprising that it was changed to 333914 in more recent data vintages.

Thank you @bl-young for your help!
